pkg/requestor/requestor.go

```diff
@@ -83,51 +85,73 @@ func (r *Requestor) consumeMessages() error {
 				done <- fmt.Errorf("failed to unmarshal as message %w", err1)
 				return
 			}
-			if r.jenkinsBuild == -1 && m.IsBuild() {
-				klog.V(2).Infof("received build message")
-				bm := messages.NewBuildMessage(-1)
-				err1 = json.Unmarshal(d.Body, bm)
-				klog.V(4).Infof("received build message %#v", bm)
-				if err1 != nil {
-					done <- fmt.Errorf("failed to unmarshal as build message %w", err1)
-					return
-				}
-				r.jenkinsBuild = bm.Build
-				fmt.Printf("Following jenkins build %d\n", r.jenkinsBuild)
-			} else if r.jenkinsBuild == m.Build {
-				if m.ISLog() {
-					klog.V(2).Infof("received log message")
-					lm := messages.NewLogsMessage(-1, "")
-					err1 = json.Unmarshal(d.Body, lm)
-					if err1 != nil {
-						done <- fmt.Errorf("failed to unmarshal as logs message %w", err1)
-						return
-					}
-					klog.V(4).Infof("log message %#v", lm)
-					fmt.Println(lm.Logs)
-				} else if m.IsStatus() {
-					klog.V(2).Infof("received status message")
-					sm := messages.NewStatusMessage(-1, false)
-					err1 = json.Unmarshal(d.Body, sm)
-					if err1 != nil {
-						done <- fmt.Errorf("failed to unmarshal as status message %w", err1)
+			//process message only if the jenkins projects match
+			if m.JenkinsProject == r.jenkinsProject {
```

Comment:
Will this also stop the condition in which test are triggered on the wrong project?

Reply:
Not the condition where tests are triggered on the wrong project but it will certainly ensure logs from the wrong project are not processed by the requestor. 

Basically, if a wrong project situation occurs (which should not as queues are separate, unless args were passed incorrectly) this time, it will timeout rather than show wrong logs

Reply:
https://github.com/mohammedzee1000/ci-firewall/pull/15/commits/abc6edc51fd5e25894d4e7ee1dde67e2e57e4f71 should ensure build does not get triggered at all if project name is mismatched